experiment/measurer/experiment_coverage_utils.py

```diff
@@ -0,0 +1,189 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Utility DataFrameContainer class and functions for report generation on
+segment and function coverage for the entire experiment
+(all Fuzzer-benchmark-trial combinations)."""
+
+import os
+import hashlib
+import pandas as pd
+
+from common import experiment_path as exp_path
+from common import filestore_utils
+from experiment.measurer import coverage_utils
+
+
+class DataFrameContainer:  # pylint: disable=too-many-instance-attributes
+    """Maintains segment and function coverage information, and writes this
+    information to CSV files."""
+
+    def __init__(self):
+        """Constructor"""
+        self.segment_entries = []
+        self.function_entries = []
+        self.fuzzer_names = dict()
+        self.benchmark_names = dict()
+        self.function_names = dict()
+        self.file_names = dict()
+
+        # Will be initialized upon done_adding_entries().
+        self.segment_df = None
+        self.function_df = None
+        self.name_df = None
+
+    def add_function_entry(self, benchmark, fuzzer, trial_id, function,
+                           function_hits, time):
+        # pylint: disable=too-many-arguments
```

Comment:
I think you want to put this on the line with the arguments (you may need to run lint and format a few times to get it right, lint and format have some weird disagreement about this but eventually get to the point where both agree).